bindings/wasm/src/credential/credential.rs

```diff
@@ -195,18 +196,25 @@ impl WasmCredential {
     self.0.non_transferable
   }
 
-  /// Returns a copy of the proof used to verify the {@link Credential}.
+  /// Optional cryptographic proof, unrelated to JWT.
   #[wasm_bindgen]
-  pub fn proof(&self) -> Result<JsValue> {
-    // TODO: Update with proof.
-    JsValue::from_serde(&self.0.proof).wasm_result()
+  pub fn proof(&self) -> Option<WasmProof> {
+    self.0.proof.clone().map(WasmProof)
   }
 
   /// Returns a copy of the miscellaneous properties on the {@link Credential}.
   #[wasm_bindgen]
   pub fn properties(&self) -> Result<MapStringAny> {
     MapStringAny::try_from(&self.0.properties)
   }
+
+  /// Sets the proof property of the {@link Credential}.
+  ///
+  /// Note that this proof is not related to JWT.
+  #[wasm_bindgen(js_name = "setProof")]
+  pub fn set_proof(&mut self, proof: Option<WasmProof>) {
```

Comment:
We don't have setters for any other property, why for proof?
We expect them to be set in the constructor. Why not add it to `ICredential`?

Reply:
well, the way I thought a proof can be made is by constructing the credential without proof, creating the signature of the resulted JSON or whatever and then you set the proof. So, it's important to have a credential without proof and then be able to set the proof. Also, the credential builder in Rust doesn't allow to set the proof.

Reply:
Fair point. Let's keep the setter. Still we should add it to `ICredential`, just to have the option I think. One of the examples in the spec that can be set as the `proof` was a proof-of-work, which may or may not be independent of the Credential. Then it's more convenient to set it all in the constructor right away.